src/ptf/dataplane.py

```diff
@@ -49,23 +49,38 @@
 #except:
 #    pass
 
+dump_pkts_on_mismatch = False
+
 def match_exp_pkt(exp_pkt, pkt):
     """
     Compare the string value of pkt with the string value of exp_pkt,
     and return True iff they are identical.  If the length of exp_pkt is
     less than the minimum Ethernet frame size (60 bytes), then padding
     bytes in pkt are ignored.
     """
+    is_match = False
     if isinstance(exp_pkt, mask.Mask):
         if not exp_pkt.is_valid():
+            logging.debug("match_exp_pkt: exp_pkt is not valid!")
             return False
-        return exp_pkt.pkt_match(pkt)
-    e = str(exp_pkt)
-    p = str(pkt)
-    if len(e) < 60:
-        p = p[:len(e)]
-    return e == p
-
+        is_match = exp_pkt.pkt_match(pkt)
+    else:
+        e = str(exp_pkt)
+        p = str(pkt)
+        if len(e) < 60:
+            p = p[:len(e)]
+        is_match = (e == p)
+
+    if (dump_pkts_on_mismatch and not is_match):
```

Comment:
I think dump_pkt can be a parameter to this function, so that callers can
set it as needed. There are wrapper functions that use this function where
pkts are mis-matched as a valid case and therefore we should not keep
printing them.

One time config option is not good for these cases. Global may be ok .. but
I prefer parameter to this function.

On Thu, Jul 28, 2016 at 1:35 PM, JK Lee <notifications@github.com> wrote:

> In src/ptf/dataplane.py
> <https://github.com/p4lang/ptf/pull/27#discussion_r72697453>:
>
> > -        return exp_pkt.pkt_match(pkt)
> > -    e = str(exp_pkt)
> > -    p = str(pkt)
> > -    if len(e) < 60:
> > -        p = p[:len(e)]
> > -    return e == p
> > -
> > +        is_match = exp_pkt.pkt_match(pkt)
> > +    else:
> > +        e = str(exp_pkt)
> > +        p = str(pkt)
> > +        if len(e) < 60:
> > +            p = p[:len(e)]
> > +        is_match = (e == p)
> > +
> > +    if (dump_pkts_on_mismatch and not is_match):
>
> There are multiple locations (at least 4, spread over in ptf-utils and
> ptf) where this function is called and there would be more usecases. Isn't
> it better to have a capability of dumping in this function?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/p4lang/ptf/pull/27/files/e20527025c937cc9140a015ae904e12339fd908e#r72697453>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKpphrug83O1zMCA9R5FcA1FQjkF3gokks5qaRKlgaJpZM4JXfik>
> .
>